src/main/java/com/synopsys/integration/jenkins/scan/service/bitbucket/BitBucketRepositoryService.java

```diff
@@ -0,0 +1,128 @@
+        this.listener = listener;
+        this.envVars = envVars;
+    }
+
+
+    public BitBucket fetchBitbucketRepoDetails(Map<String, Object> scanParameters) throws IOException, InterruptedException {
+
+        // extracting the job name from the combined job_branch name
+        String jobName = envVars.get("JOB_NAME").substring(0, envVars.get("JOB_NAME").indexOf("/"));
+
+        //getting the pull number from env
+        Integer projectRepositoryPullNumber = envVars.get("CHANGE_ID") != null ? Integer.valueOf(envVars.get("CHANGE_ID")) : 0;
+
+        listener.getLogger().println("Getting bitbucket repository details");
+
+        BitBucket bitBucket = new BitBucket();
+
+        Jenkins jenkins = Jenkins.getInstanceOrNull();
+
+        if (jenkins != null) {
+
+            SCMSource scmSource = findSCMSource(jenkins, jobName);
+
+            if (scmSource instanceof BitbucketSCMSource) {
+                BitbucketSCMSource bitbucketSCMSource = (BitbucketSCMSource) scmSource;
+
+                // Creating and using them to get the bitbucket token
+                FreeStyleProject freeStyleProject = new FreeStyleProject(jenkins, jobName);
+                FreeStyleBuild freeStyleBuild = new FreeStyleBuild(freeStyleProject);
+
+                // If we are not able to get bitbucket token from the input directly then we will get it through the bitbucketSCMSource CredentialId
+                String bitBucketToken = (String) scanParameters.get("bitbucket_token");
+                if(bitBucketToken == null) {
+                    bitBucketToken = getCredentialsToken(bitbucketSCMSource.getCredentialsId(), freeStyleBuild);
+                }
+
+                BitbucketApi bitbucketApi = bitbucketSCMSource.buildBitbucketClient(bitbucketSCMSource.getRepoOwner(), bitbucketSCMSource.getRepository());
+
+                // Access the repository details with BitbucketApi
+                BitbucketRepository bitbucketRepository = bitbucketApi.getRepository();
+
+                listener.getLogger().println("Repository Name: " + bitbucketRepository.getRepositoryName());
+
+                Api bitBucketApi = new Api();
+                bitBucketApi.setUrl(bitbucketSCMSource.getServerUrl());
+                bitBucketApi.setToken(bitBucketToken);
+
+                Pull pull = new Pull();
+                pull.setNumber(projectRepositoryPullNumber);
+
+                Repository repository = new Repository();
+                repository.setName(bitbucketRepository.getRepositoryName());
+                repository.setPull(pull);
+
+                Project project = new Project();
+                project.setKey(bitbucketRepository.getProject().getKey());
+                project.setRepository(repository);
+
+                bitBucket.setApi(bitBucketApi);
+                bitBucket.setProject(project);
+
+            } else {
+                listener.getLogger().println("SCM source is not a BitbucketSCMSource.");
```

Comment:
Ignoring bitbucket_automation_fixpr and  bitbucket_automation_prcomment since couldn't find any valid Bitbucket SCM source.